src/pages/Search.jsx

```diff
@@ -1,7 +1,95 @@
+    const fetchSearchResults = async () => {
+      try {
+        // API to fetch search results and set them to the results and the totalpage state.
+        const response = await api.get(
+          `/products/search?k=${keyword}&c=${categorySlug}&page=${currentPage}&limit=${itemsPerPage}`
+        );
+        setResults(response.data.results);
+        setTotalPages(response.data.totalPages);
+      } catch (error) {
+        toast.error('Error fetching search results', error);
+      }
+    };
+    // When either keyword or categorySlug is changed, or when currentPage is changed, the fetchSearchResults() is called to retrieve and display new search results.
+    // However, if both keyword and categorySlug are not present, meaning that search criteria are not provided.
+    if (keyword || categorySlug) {
+      fetchSearchResults();
+    }
+  }, [keyword, categorySlug, currentPage]);
+
   return (
-    <div>
-      <p>This is the Search page</p>
-    </div>
+    <Layout>
+      <div className='container mx-auto px-6'>
+        <h3 className='text-gray-700 text-4xl font-medium'>
+          {keyword ? `Products for "${keyword}"` : `Products in category "${categorySlug}"`}
+        </h3>
+        <div className='grid gap-6 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 mt-6'>
+          {/* Checks if there　is at least one or more elements in the results array. If results contain items, it maps over each product*/}
+          {results.length > 0 ? (
+            results.map((product) => (
+              <div
+                key={product._id}
+                className='w-full min-w-[240px] mx-auto rounded-md shadow-md overflow-hidden'
+              >
+                <Link to={`/product/${product.slug}`}>
+                  <div className='flex flex-col w-full bg-cover items-center '>
+                    <Image
+                      alt='product image'
+                      className='h-40 max-w-full object-cover rounded-t-lg '
+                      src={product.imageUrl}
+                    />
+                    <div className='px-5 py-3 flex flex-col justify-end mt-auto '>
+                      <div className='text-center font-medium'>
+                        <h3 className='font-bold text-gray-900 uppercase mb-1'>{product.name}</h3>
+                        <p className=' text-gray-500'>{product.category.name}</p>
+                      </div>
+                    </div>
+                  </div>
+                  <div className='mb-3 mr-3'>
+                    <p className='font-semibold text-gray-900 text-lg text-right'>
+                      Price: ${product.price}
+                    </p>
+                  </div>
+                </Link>
+              </div>
+            ))
+          ) : (
+            // If results is empty, it displays a message below.
+            <p>No products found for '{keyword}'</p>
+          )}
+        </div>
+        <div className='flex justify-center mt-7'>
+          <Pagination
```

Comment:
Maybe we can think about removing the pagination when there is only one page or none at all? This could really streamline the interface, especially in cases with limited content. Just a small idea to ponder!

However, don't worry if it seems a bit excessive at the moment. This isn't a crucial feature, so feel free to come back to it whenever you have some spare time.

Here the suggestion from ChatGPT
![Screenshot 2023-12-20 at 11 31 24 pm](https://github.com/irene2mana-T3A2-2023/SakuraPantryClient/assets/98362650/13c0728d-9240-4f34-9f67-2d4a82681c96)